lib/ex.coffee

```diff
@@ -1,9 +1,30 @@
 class Ex
-  write: ->
+  quit: ->
+    atom.workspace.getActivePane().destroyActiveItem()
+  q: =>
+    @quit()
+  tabedit: (filenames) ->
+    if filenames? and filenames.length > 0
+      atom.open(pathsToOpen: filenames)
+    else
+      atom.open()
+  tabe: (filenames) =>
+    @tabedit(filenames)
+  write: (close=false) =>
```

Comment:
Seems like mixing user supplied parameters (`:tabe <filename>`) and internal parameters like this `close` might lead to problems. Maybe instead any potentially asyncronous operations here should return promises.

One possible implementation might look like this.

```CoffeeScript
    write: =>
        dfd = Promise.defer()
        if atom.workspace.getActiveTextEditor().getPath() isnt undefined
            # not sure of the exact implementation. Just trying to put this together in a github comment.
        else
            atom.workspace.getActivePane().saveActiveItemAs(dfd.resolve)
        dfd.promise
        

    wq: =>
        @write().then(=> @quit())
````